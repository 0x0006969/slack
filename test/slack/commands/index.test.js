process.env.SLACK_VERIFICATION_TOKEN = 'test';

const request = require('supertest');
const createProbot = require('probot');
const nock = require('nock');

const slack = require('../../../lib/slack');

const repoFixture = require('../../fixtures/repo');
let commandFixture = require('../../fixtures/slack/command.subscribe');

describe('commands', () => {
  let probot;

  beforeEach(() => {
    nock.enableNetConnect(/127\.0\.0\.1/);

    probot = createProbot({});
    probot.load(slack);
  });

  describe('/github subscribe https://github.com/owner/repo', () => {
    test('confirms the subscription', async () => {
      const scope = nock('https://api.github.com').get('/repos/atom/atom')
        .reply(200, repoFixture);

      const req = request(probot.server).post('/slack/command').send(commandFixture);

      await req.expect(200).expect((res) => {
        expect(res.body).toMatchSnapshot();
      });

      expect(scope.isDone()).toBe(true);
    });
  });

  describe('/github subscribe wat?', () => {
    test('shows an error', async () => {
      commandFixture = { ...commandFixture, text: 'subscribe wat?' };

      const req = request(probot.server).post('/slack/command').send(commandFixture);

      await req.expect(200).expect((res) => {
        expect(res.body).toMatchSnapshot();
      });
    });
  });
});
