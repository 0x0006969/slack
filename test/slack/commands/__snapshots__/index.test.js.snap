// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands /github subscribe https://github.com/owner/repo confirms the subscription 1`] = `
Object {
  "attachments": Array [
    Object {
      "response_type": "in_channel",
      "text": "Subscribed <#C2147483705> to <https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
    },
  ],
}
`;

exports[`commands /github subscribe wat? shows an error 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "danger",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "\`wat?\` does not appear to be a GitHub link.",
    },
  ],
}
`;
